runtime/rasdump/dmpsup.c

```diff
@@ -1367,24 +1371,40 @@ J9VMDllMain(J9JavaVM *vm, IDATA stage, void *reserved)
 			if (OMR_ERROR_NONE != initDumpDirectory(vm)) {
 				retVal = J9VMDLLMAIN_FAILED;
 			} else {
-				/* Defer enabling dump agents until initRasDumpGlobalStorage() is finished
-				 * so as to avoid triggering dump agent before RAS is ready for use.
-				 */
-				initRasDumpGlobalStorage(vm);
-
-				/* Swap in new dump facade */
-				if (OMR_ERROR_NONE == pushDumpFacade(vm)) {
-					retVal = configureDumpAgents(vm);
-					/* Allow configuration updates now that we've done initial setup */
-					unlockConfig();
-				} else {
-					freeRasDumpGlobalStorage(vm);
-					retVal = J9VMDLLMAIN_FAILED;
-				}
+//				/* Defer enabling dump agents until initRasDumpGlobalStorage() is finished
+//				 * so as to avoid triggering dump agent before RAS is ready for use.
+//				 */
+//				initRasDumpGlobalStorage(vm);
+//
+//				/* Swap in new dump facade */
+//				if (OMR_ERROR_NONE == pushDumpFacade(vm)) {
+//					retVal = configureDumpAgents(vm);
+//					/* Allow configuration updates now that we've done initial setup */
+//					unlockConfig();
+//				} else {
+//					freeRasDumpGlobalStorage(vm);
+//					retVal = J9VMDLLMAIN_FAILED;
+//				}
```

Comment:
it is for disabling OOM dumps by default only for gcpolicy:nogc, comment out the code is for initializing dumps settings( both default and explicit options), I defer the code from PORT_LIBRARY_GUARANTEED stage(J9VMDllMain) to ALL_LIBRARIES_LOADED stage(J9VMDllMain) due to gcpolicy is parsed in  ALL_LIBRARIES_LOADED stage. it might generate minor behavior changes(no dump function before initRasDumpGlobalStorage()), so I still try to find another way try not to move the code.